tests/keras_tests/feature_networks_tests/feature_networks/qat/qat_test.py

```diff
@@ -121,3 +116,72 @@ def compare(self, quantized_model, float_model, loaded_model, input_x=None, quan
+
+    def get_tpc(self):
+        return get_tpc("QAT_wrappers_test", weight_bits=self.weight_bits, activation_bits=self.activation_bits,
+                       weights_quantization_method=self.weights_quantization_method,
+                       activation_quantization_method=self.activation_quantization_method)
+
+    def create_networks(self):
+        inputs = layers.Input(shape=self.get_input_shapes()[0][1:])
+        outputs = self.layer(inputs)
+        return keras.Model(inputs=inputs, outputs=outputs)
+
+    def run_test(self, experimental_facade=False, **kwargs):
+        model_float = self.create_networks()
+        ptq_model, quantization_info, custom_objects = mct.keras_quantization_aware_training_init(model_float,
+                                                                                                  self.representative_data_gen,
+                                                                                                  fw_info=self.get_fw_info(),
+                                                                                                  target_platform_capabilities=self.get_tpc())
+        qat_model = ptq_model
+        if self.test_loading:
+            qat_model.save('qat2model.h5')
+            qat_model = mct.keras_load_quantized_model('qat2model.h5')
+
+        self.compare(qat_model,
+                     finalize=False,
+                     input_x=self.representative_data_gen(),
+                     quantization_info=quantization_info)
+
+        if self.finalize:
+            qat_model = mct.keras_quantization_aware_training_finalize(qat_model)
+            self.compare(qat_model,
+                         finalize=True,
+                         input_x=self.representative_data_gen(),
+                         quantization_info=quantization_info)
+
+    def compare(self, qat_model, finalize=False, input_x=None, quantization_info=None):
+
+        for layer in qat_model.layers:
+            if isinstance(layer, qi.KerasQuantizationWrapper):
+                # Check Activation quantizers
+                if layer.dispatcher.is_activation_quantization:
+                    for quantizer in layer.dispatcher.activation_quantizers:
+                        if finalize:
+                            self.unit_test.assertTrue(isinstance(quantizer, qi.BaseKerasInferableQuantizer))
+                        else:
+                            self.unit_test.assertTrue(isinstance(quantizer, qi.BaseKerasTrainableQuantizer))
+
+                # Check Weight quantizers
+                if layer.dispatcher.is_weights_quantization:
+                    for name, quantizer in layer.dispatcher.weight_quantizers.items():
+                        if finalize:
+                            self.unit_test.assertTrue(isinstance(quantizer, qi.BaseKerasInferableQuantizer))
+                        else:
+                            self.unit_test.assertTrue(isinstance(quantizer, qi.BaseKerasTrainableQuantizer))
```

Comment:
Maybe we should check also the the inferable params are the expected params according to the trainable params

Reply:
Maybe. The only problem is that it now depends on specific quantizer and maybe it's belong to the quantizer test itself.